logger/logger.go

```diff
@@ -86,6 +87,20 @@ func LogrusLogLevelFrom(configLogLevel string) logrus.Level {
 	return logLevel
 }
 
+func backtrace() string {
+	var filenames []string
+
+	pc := make([]uintptr, 1000)
+	n := runtime.Callers(0, pc)
+	frames := runtime.CallersFrames(pc[:n])
+
+	for frame, isNextFrameValid := frames.Next(); isNextFrameValid; frame, isNextFrameValid = frames.Next() {
+		filenames = append(filenames, fmt.Sprintf("%s:%d", frame.File, frame.Line))
+	}
```

Comment:
In this case, why is it needed to have `frame, isNextFrameValid := frames.Next(); isNextFrameValid; frame, isNextFrameValid = frames.Next()` inlined? Is it because we need a first call to `.Next`, check that it is valid, and then another one?

If that's the case, what do you think about having it like follows?

```go
var frame runtime.Frame
var isNextFrameValid bool

// Get the first frame to start the loop.
frame, isNextFrameValid = frames.Next()
for isNextFrameValid {
	filenames = append(filenames, fmt.Sprintf("%s:%d",frame.File, frame.Line))

	frame, isNextFrameValid = frames.Next()
}
```

Reply:
yes and it was because of it. I don't have preference here so I changed it as you suggested :)

Reply:
Well, I might have to apologize here. I didn't see it was like a simple `i := 0; i < something; i++` loop, and initially I thought "this might be skipping loops?" :man_facepalming: 

I might have made you change this just because I didn't understand it, sorry :sweat_smile: . It's okay if you revert it back: as you want :+1:

Reply:
I changed it back as that loop allows put everything inside , thanks :)